runtime/platform.c

```diff
@@ -209,23 +211,29 @@ void caml_mem_decommit(void* mem, uintnat size)
 
 void caml_mem_unmap(void* mem, uintnat size)
 {
-#ifdef DEBUG
-  uintnat data;
-  CAMLassert(caml_lf_skiplist_find(&mmap_blocks, (uintnat)mem, &data) != 0);
-  CAMLassert(data == size);
-#endif
   caml_gc_message(0x1000, "munmap %" ARCH_INTNAT_PRINTF_FORMAT "d"
                           " bytes at %p for heaps\n", size, mem);
   caml_plat_mem_unmap(mem, size);
-#ifdef DEBUG
-  caml_lf_skiplist_remove(&mmap_blocks, (uintnat)mem);
-#endif
 }
 
 #define Min_sleep_ns       10000 // 10 us
 #define Slow_sleep_ns    1000000 //  1 ms
 #define Max_sleep_ns  1000000000 //  1 s
 
+/* Again, TODO: how does this work on mingw-w64 atm? */
```

Comment:
Mingw-w64 emulation is at:

https://github.com/mirror/mingw-w64/blob/master/mingw-w64-crt/misc/mingw_usleep.c

If I understand correctly, we would include some equivalent code to have it available for MSVC as well?

Reply:
Thanks for digging! TIL there are many more emulated CRT functions in mingw-w64! Indeed, this looks like something we'd want to move to win32.c as `caml_usleep` and then have `osdeps.h` (I think?) deal with it.

In the absence of benchmarks, I'm not sure what's worse - `usleep` in mingw-w64 is actually "millisecond sleep", whereas I'm not sure that the function I put in platform.c (which I very quickly grabbed from somewhere!) making that many system calls is going to be particularly good at sub-ms granularity). On the other hand, this is used for `SPIN_WAIT` throughout the gc.

Reply:
The most precise way to sleep on Windows is this: https://stackoverflow.com/questions/7827062/is-there-a-windows-equivalent-of-nanosleep

Reply:
Thanks @XVilka. though the issue is not the _precision_ but the _granularity_.